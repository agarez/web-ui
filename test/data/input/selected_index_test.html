<!doctype html>
<!--
Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
</head>
<body>
  Test a two-way binding on selected-index. It will programmatically select
  the second option, "Night Owl":
  <select id="awake" on-change="checkit($event)" bind-selected-index="awake">
    <option template repeat="x in values" value="{{x}}">{{x}}</option>
  </select>
  <script type="application/dart">
    @observable library main;
    import 'dart:async';
    import 'dart:html';
    import 'package:unittest/unittest.dart';
    import 'package:web_ui/web_ui.dart';

    var values = ['Early Bird', 'Night Owl'];
    int awake = 0;

    void main() {
      SelectElement e = query('#awake');
      Timer.run(() {
        awake = 2;
        Timer.run(() {
          expect(awake, 2);
          expect(e.selectedIndex, 2);
          expect(e.value, values[1]);

          window.postMessage('done', '*');
        });
      });
    }
  </script>
</body>
</html>
