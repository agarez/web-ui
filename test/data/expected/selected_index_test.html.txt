Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x60
  RenderBlock {HTML} at (0,0) size 800x60
    RenderBody {BODY} at (8,8) size 784x44
      RenderText {#text} at (0,0) size 770x40
        text run at (0,0) width 740: "Test a two-way binding on selected-index. It will programmatically select "
        text run at (740,0) width 30: "the"
        text run at (0,22) width 280: "second option, \"Night Owl\": "
      RenderMenuList {SELECT} at (282,20) size 125x22 [bgcolor=#DDDDDD] [border: (1px solid #000000)]
        RenderBlock (anonymous) at (1,1) size 123x20
          RenderText at (4,1) size 90x18
            text run at (4,1) width 90: "Night Owl"
      RenderText {#text} at (0,0) size 0x0
      RenderText {#text} at (0,0) size 0x0
